engine/benches/bench.rs

```diff
@@ -21,7 +21,13 @@ fn lowercase<'a>(args: &mut dyn Iterator<Item = LhsValue<'a>>) -> LhsValue<'a> {
     let input = args.next().unwrap();
     match input {
         LhsValue::Bytes(mut bytes) => {
-            bytes.to_mut().make_ascii_lowercase();
+            let make_lowercase = match bytes {
+                Cow::Borrowed(bytes) => bytes.iter().any(u8::is_ascii_uppercase),
```

Comment:
I know lowercase/uppercase are properly defined for ascii bytes, but i would rather express this as:
`any(|b| !u8::is_ascii_lowercase(b))`

And equivalent for `uppercase` function

Reply:
It's not the same though! (I almost did the same mistake)

One checks whether there is any byte that is uppercase, while another checks whether there is any character that is not lowercase.

For example, take a string: `1234` - `any(u8::is_ascii_uppercase)` will return false because it doesn't have any uppercase chars, but `any(|b| !u8::is_ascii_lowercase(b))` will return true because it has characters that are not lowercase chars (they're digits instead).

Reply:
Ah ok, my bad.